File: pe/visualstudio/PE.CSharp/Problem0001.cs
using System.Linq;

namespace PE.CSharp
{
    public class Problem0001
    {
        public int SumOfMultiplesOf3And5(int maximumNumberExclusive)
        {
            return Enumerable
                .Range(1, maximumNumberExclusive - 1)
                .Where(IsAMultipleOf3Or5)
                .Sum();
        }

        public bool IsAMultipleOf3Or5(int number)
        {
            return IsAMultipleOf3(number) || IsAMultipleOf5(number);
        }

        private bool IsAMultipleOf5(int number)
        {
            return number % 5 == 0;
        }

        private bool IsAMultipleOf3(int number)
        {
            return number% 3 == 0;
        }
    }
}

File: pe/visualstudio/PE.JavaScript/Implementation/Problem0001.js
var javascriptPE;
(function (javascriptPE) {
    (function(implementation) {

        var problem0001 = (function () {

            function problem0001(listHelperLibrary) {
                this.listHelpers = listHelperLibrary;
            }

            problem0001.prototype.isAMultipleOf3Or5 = function(number) {
                return number % 3 === 0 || number % 5 === 0;
            };

            problem0001.prototype.sumOfMultiplesOf3And5 = function(maximumNumberExclusive) {
                var listOfPossibleMultiples = this.listHelpers.generateSequentialArray(1, maximumNumberExclusive - 1);
                var filteredList = listOfPossibleMultiples.filter(this.isAMultipleOf3Or5);
                return this.listHelpers.sum(filteredList);
            };

            return problem0001;
        })(); 
        implementation.problem0001 = problem0001;

    })(javascriptPE.implementation || (javascriptPE.implementation = {}));
})(javascriptPE || (javascriptPE = {}));